homework/list.js

```diff
@@ -1,79 +1,126 @@
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    /**
+    * Fetch remote data to render list and
+    * bind 'click' event to initialize the ListManager.
+    *
+    * @this {ListManager}
+    */
+    start() {
+      this.fetchList()
+      .then((function(data) {
+        this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();
+      }).bind(this) )
+      .catch((function(){
+        //exception handle code here
+      }).bind(this) );
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+      window.addEventListener('click', (function(event) {
+        this.onNoteOpen(event);
+      }).bind(this));
+    },
 
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    /**
+    * Dispatch a 'note-open' event if event target has 'note-title' in its classList.
+    *
+    * @param {CustomEvent} event Its target may have 'note-title' in classList.
+    * @this {ListManager}
+    */
+    onNoteOpen(event) {
+      if (event.target.classList.contains('note-title')) {
+        var id = event.target.dataset.noteId;
+        var content = this._listNoteContent[id];
+        window.dispatchEvent(new CustomEvent('note-open',
+          { detail: content }));
+      };
+    },
 
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
+    /**
+    * Dispatch a 'note-open' event with first note if note list is not empty
+    *
+    * @this {ListManager}
+    */
+    preloadFirstNote() {
+      if (this._listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped with one conditional statement with only one branch, the `if...else` should be put in the caller, rather than the callee.